### ISSUE TITLE
CA2000 possible false positive

### ISSUE THREAD

**Issue by zwergenaufstand** [dotnet/roslyn-analyzers#7527]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 9.0.101](https://dotnet.microsoft.com/download/dotnet/9.0)

### Describe the bug

The following program (see [Steps To Reproduce](#steps-to-reproduce)) triggers CA2000 even though all disposable objects are disposed.

### Steps To Reproduce

1. Compile the following program targeting .NET Framework 4.8.

Ca2000FalsePositiveTestCase.csproj:
```
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net48</TargetFramework>
    <OutputType>Exe</OutputType>
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisLevel>latest-all</AnalysisLevel>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <WarningLevel>4</WarningLevel>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
</Project>
```

Program.cs:
```
using System;
using System.Drawing;

namespace Ca2000FalsePositiveTestCase
{
    internal static class Program
    {
        static void Main()
        {
            using (_ = new Bitmap(1, 1))
            {
                for (int i = 0; i < 1; ++i)
                {
                    using (var bmp = new Bitmap(1, 1))
                    {
                        if (bmp.Tag is null)
                            throw new InvalidOperationException();

                        new Bitmap(1, 1).Dispose();
                    }
                }
            }
        }
    }
}
```

### Expected behavior

No CA2000 warning.

### Actual behavior

CA2000 is triggered:

```
Program.cs(14,38): error CA2000: Use recommended dispose pattern to ensure that object created by 'new Bitmap(1, 1)' is disposed on all paths. If possible, wrap the creation within a 'using' statement or a 'using' declaration. Otherwise, use a try-finally pattern, with a dedicated local variable declared before the try region and an unconditional Dispose invocation on non-null value in the 'finally' region, say 'x?.Dispose()'. If the object is explicitly disposed within the try region or the dispose ownership is transfered to another object or method, assign 'null' to the local variable just after such an operation to prevent double dispose in 'finally'. (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)
```
